packages/template-retail-react-app/app/pages/login/index.test.js

```diff
@@ -123,35 +132,51 @@ test('Allows customer to sign in to their account', async () => {
         wrapperProps: {siteAlias: 'uk', locale: {id: 'en-GB'}, appConfig: mockConfig.app}
     })
 
+    expect(await screen.findByText('Auth Type:', {}, {timeout: 30000})).toBeInTheDocument()
+
     // enter credentials and submit
-    user.type(screen.getByLabelText('Email'), 'darek@test.com')
-    user.type(screen.getByLabelText('Password'), 'Password!1')
-    user.click(screen.getByText(/sign in/i))
+    await user.type(screen.getByLabelText('Email'), 'darek@test.com')
+    await user.type(screen.getByLabelText('Password'), 'Password!1')
+    await user.click(screen.getByText(/sign in/i))
 
     // wait for success state to appear
-    expect(await screen.findByText(/Welcome Back/i, {}, {timeout: 30000})).toBeInTheDocument()
-    expect(await screen.findByText(/darek@test.com/i, {}, {timeout: 30000})).toBeInTheDocument()
+    expect(
+        await screen.findByText(/Auth Type: registered/i, {}, {timeout: 30000})
+    ).toBeInTheDocument()
 })
 
 test('Renders error when given incorrect log in credentials', async () => {
     // mock failed auth request
     global.server.use(
+        rest.get('*/customers', (req, res, ctx) => {
+            return res(ctx.delay(0), ctx.status(404), ctx.json({message: 'Not Found.'}))
+        }),
         rest.post('*/oauth2/login', (req, res, ctx) =>
-            res(ctx.delay(0), ctx.status(401), ctx.json({message: 'Invalid Credentials.'}))
+            res(
+                ctx.delay(0),
+                ctx.status(401),
+                ctx.json({
+                    status_code: '401 UNAUTHORIZED',
+                    message: 'Invalid Credentials.'
+                })
+            )
         )
     )
 
+    mockAuthHelperFunctions[AuthHelpers.LoginRegisteredUserB2C].mutateAsync.mockRejectedValueOnce(
```

Comment:
We already mock the login at the api, why do we need to moch these helpers? Should these helpers proceed the response from the mock api?

Reply:
These helpers do a whole bunch of other things like decoding the token, extracting parts of the token, setting various tokens to different storages like access_token going to localStorage and refresh token going to cookies etc.

Ideally these should be tested in the commerce-sdk-react package itself.
Since the template no longer handles auth itself, the template should not be testing login flows.
It should only be testing the state of the app before and after login.

So we mock the helper to simulate a successful or failed login state and make sure the template renders to the respective state after.